### PR TITLE
Consume message from retry topic

### PR DESCRIPTION
**Pull request by SteveHicksCH**
ivp 1301

### CODE REVIEW

File: src/test/java/uk/gov/companieshouse/ocrapiconsumer/kafka/OcrApiConsumerKafkaConsumerTest.java [hunk 1]
```diff
@@ -74,19 +76,84 @@ void sentMessageToRetryTopicAfterRetryableErrorOnMainTopic() throws Serializatio
...
-        doThrow(new uk.gov.companieshouse.ocrapiconsumer.kafka.exception.RetryableErrorException("Dummy", new Exception("dummy cause")))
+        org.springframework.messaging.Message<OcrRequestMessage> message = createTestMessage(kafkaConsumer.getMainTopicName());
+        doThrow(newRetryableError())
 				.when(ocrApiConsumerService).ocrRequest(message.getPayload());
         when(serializerFactory.getGenericRecordSerializer(OcrRequestMessage.class)).thenReturn(serializer);
         when(serializer.toBinary(any())).thenReturn(new byte[4]);
 
         // When
-        kafkaConsumer.consumeOcrApiRequestMessage(message, new ConsumerRecordMetadata(getRecordMetadata(), null));
+        kafkaConsumer.consumeOcrApiRequestMessage(message, metaDatawithTopic(kafkaConsumer.getMainTopicName()));
```
Comment (csciubbach) [inline src/test/java/uk/gov/companieshouse/ocrapiconsumer/kafka/OcrApiConsumerKafkaConsumerTest.java:86]: Reconsider the function name maybe to getMetadataWithTopic (if you keep, capitalise W)

File: src/test/java/uk/gov/companieshouse/ocrapiconsumer/kafka/OcrApiConsumerKafkaConsumerTest.java [hunk 2]
```diff
@@ -119,9 +186,10 @@ public MessageHeaders getHeaders() {
         };
     }
 
-    private RecordMetadata getRecordMetadata() {
+    private ConsumerRecordMetadata metaDatawithTopic(String topicName) {
```
Comment (csciubbach) [inline src/test/java/uk/gov/companieshouse/ocrapiconsumer/kafka/OcrApiConsumerKafkaConsumerTest.java:189]: This is the biggest market of the metadata vs metaData inconsistency.

Please choose either metaData/MetaData or metadata/Metadata

File: src/test/java/uk/gov/companieshouse/ocrapiconsumer/kafka/OcrApiConsumerKafkaConsumerTest.java (hunk 1 above)
Comment (SteveHicksCH) [reply src/test/java/uk/gov/companieshouse/ocrapiconsumer/kafka/OcrApiConsumerKafkaConsumerTest.java:86]: New name metadataWithTopic

File: src/test/java/uk/gov/companieshouse/ocrapiconsumer/kafka/OcrApiConsumerKafkaConsumerTest.java (hunk 2 above)
Comment (SteveHicksCH) [reply src/test/java/uk/gov/companieshouse/ocrapiconsumer/kafka/OcrApiConsumerKafkaConsumerTest.java:189]: as above

**Review by csciubbach — APPROVED** [review]
Approved

**Pull request closed without merging**